packages/integration-objects/srv/integration-objects/simple-exchange-rate-type-service.js

```diff
@@ -0,0 +1,210 @@
+  }
+
+  async function writeAuditLogForRateTypes(req, next) {
+    let currentValues;
+    const data = req.data;
+    const type = 'exchange-rate-type';
+    const auditParams = {
+      exchangeRateType: data.exchangeRateType,
+      exchangeRateTypeDescription: data.exchangeRateTypeDescription,
+      referenceCurrencyThreeLetterISOCode: data.referenceCurrencyThreeLetterISOCode
+    };
+
+    let currentValuesIsNotEmpty = true;
+    if (Constants.CREATE_EVENT !== req.event) {
+      currentValues = await fetchCurrentAttributeValues(req, req.user.tenant, ExchangeRateTypes);
+      currentValuesIsNotEmpty = !util.isNullish(currentValues);
+    }
+
+    if (currentValuesIsNotEmpty) {
+      await writeAuditLog(req, auditParams, currentValues, type);
+    }
+    await next();
+  }
+};
+
+async function validateIdInTenant(id, ExchangeRateTypes, req, affectedRows) {
+  if (id != null) {
+    const query2 = SELECT.from(ExchangeRateTypes).where({ ID: id }).limit(1, 0);
+    const tx2 = cds.transaction(req);
+    const affectedRows2 = await tx2.run(query2);
+
+    if (affectedRows2.length) {
+      const record2 = affectedRows[0];
+      if (req.user.tenant !== record2.tenantID) {
+        logger.error('Invalid ID was accessed. It is not related to the current tenant.');
+        throw new ValidationError(RateTypeExtensionConstants.GUID_NOT_FOUND_FOR_READ, ErrorStatuses.BAD_REQUEST);
+      }
+    }
+  }
+}
+
+async function fetchExistingTexts(ExchangeRateTypeDescriptions, record, req) {
+  const query = SELECT.from(ExchangeRateTypeDescriptions)
+    .where({ locale: record.locale })
+    .and({ ID_texts: record.ID_texts })
+    .limit(1, 0);
+
+  const tx = cds.transaction(req);
+  const affectedRows = await tx.run(query);
+  return affectedRows;
+}
+
+async function fetchExistingRateTypeDesc(ExchangeRateTypeDescriptions, record, req) {
+  const query = SELECT.from(ExchangeRateTypeDescriptions)
+    .where({ locale: record.locale })
+    .and({ ID: record.ID })
+    .limit(1, 0);
+
+  const tx = cds.transaction(req);
+  const affectedRows = await tx.run(query);
+  return affectedRows;
+}
+
+function checkForDuplicateRateTypeObjects(affectedRows, req) {
+  if (affectedRows.length) {
+    if (req.event === Constants.CREATE_EVENT) {
+      logger.error('Record found in an Active entity for CREATE event. The primary keys are not unique.');
+      throw new ValidationError(
+        RateTypeExtensionConstants.UNIQUE_CONSTRAINT_FOR_SEMANTIC_KEYS_EX_RATE_TYPE,
+        ErrorStatuses.BAD_REQUEST
+      );
+    } else {
+      const payloadId = req.data.ID;
+      const recordId = affectedRows[0].ID;
+      if (payloadId !== recordId) {
+        logger.error('Record found in Active Entity for UPDATE. The primary keys are not unique.');
+        throw new ValidationError(
+          RateTypeExtensionConstants.UNIQUE_CONSTRAINT_FOR_SEMANTIC_KEYS_EX_RATE_TYPE,
+          ErrorStatuses.BAD_REQUEST
+        );
+      }
+    }
+  }
```

Comment:
You don't need the else here:
```suggestion
    }
    const payloadId = req.data.ID;
    const recordId = affectedRows[0].ID;
    if (payloadId !== recordId) {
      logger.error('Record found in Active Entity for UPDATE. The primary keys are not unique.');
      throw new ValidationError(
        RateTypeExtensionConstants.UNIQUE_CONSTRAINT_FOR_SEMANTIC_KEYS_EX_RATE_TYPE,
        ErrorStatuses.BAD_REQUEST
      );
    }
  }
```